Automattic/wp-calypso
# Newsletters: Update Stripe link on Launchpad

## Proposed Changes

This PR updates Launchpad's 'Set up payment method' task to link directly to Stripe rather than to the Earn page.
   - On Launchpad, we fetches the stripe url from the memberships endpoint and link to that if available. In the API request, we set source to be 'launchpad'.
   - After connecting Stripe a user is still returned to the Earn page, but 'stripe_connect_success=launchpad` is added as a query parameter, and when that is present, we redirect back to Launchpad. 

The approach used in this PR - setting the 'source', which becomes a query param when returned from stripe, which triggers a redirect - is patterned off a prior PR [here](https://github.com/Automattic/wp-calypso/pull/75437).

## Testing Instructions

1) Setup
   - Add `define( 'USE_STORE_SANDBOX', true );` to your sandbox. You can add it to wp-content/mu-plugins/0-sandbox.php. This will allow you to NOT have to fully set up stripe when sent to stripe.
   - Sandbox public-api.wordpress.com and the domain of your test site.
   - Checkout this calypso branch and run yarn or yarn start if needed. 
   
2) Test the new link.
   - Start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro. Be sure to choose the "Paid" option. And remember to sandbox this domain before clicking the stripe link if you've created a new site. 
   - Click the 'Set up payment method' link and confirm you are taken directly to Stripe, not the Earn page.
   - If using USE_STORE_SANDBOX, you should see a link to skip the payment setup. Click that to return to WordPress.
   - You should be returned to the Earn page. Confirm that you see `?stripe_connect_success=launchpad` appended to the url.
   - Note the redirect will not work because Stripe sends you back to WordPress rather than Calypso localhost. Without changing anything else on the Earn page, replace `https://wordpress.com` with `http://calypso.localhost:3000/`. Confirm that when you do, you are redirected back to Launchpad, and the 'Set up payment method' task is complete. 
   
<img width="1112" alt="stripe" src="https://github.com/Automattic/wp-calypso/assets/21228350/02b2e509-270f-408f-a757-ccf0940100de">

 
3) Check unit tests.
   - I had to update tests. Confirm that launchpad tests still pass by running `yarn run test-client client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/`.
